examples/framesjs-starter/app/examples/multi-protocol/page.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  FrameButton,
+  FrameContainer,
+  FrameImage,
+  FrameReducer,
+  NextServerPageProps,
+  getFrameMessage,
+  getPreviousFrame,
+  useFramesReducer,
+} from "frames.js/next/server";
+import { getXmtpFrameMessage, isXmtpFrameActionPayload } from "frames.js/xmtp";
+import Link from "next/link";
+import { DEBUG_HUB_OPTIONS } from "../../debug/constants";
+import { ClientProtocolId } from "frames.js";
+
+type State = {
+  pageIndex: number;
+};
+
+const totalPages = 5;
+const initialState: State = { pageIndex: 0 };
+
+const reducer: FrameReducer<State> = (state, action) => {
+  const buttonIndex = action.postBody?.untrustedData.buttonIndex;
+
+  return {
+    pageIndex: buttonIndex
+      ? (state.pageIndex + (buttonIndex === 2 ? 1 : -1)) % totalPages
+      : state.pageIndex,
+  };
+};
+
+const acceptedProtocols: ClientProtocolId[] = [
+  {
+    id: "xmtp",
+    version: "vNext",
+  },
+  {
+    id: "farcaster",
+    version: "vNext",
+  },
+];
+
+// This is a react server component only
+export default async function Home({
+  params,
+  searchParams,
+}: NextServerPageProps) {
+  const previousFrame = getPreviousFrame<State>(searchParams);
+  const [state] = useFramesReducer<State>(reducer, initialState, previousFrame);
+
+  if (
+    previousFrame.postBody &&
```

Comment:
can we make this example be smoother than a big if else? Perhaps both use the ethereum address, they just derive it via different means and render the same thing